File: Barbara Kosior/35. Search Insert Position.cpp
class Solution {
public:
    int searchInsert(vector<int>& nums, int target) {
         
        int i=0;
        int b = nums.size()-1;

        while (i<b){
            int srednia = (i+b)/2;
            if (nums[srednia] == target)
            {
                return srednia;
            }
            else if (nums[srednia] > target)
            {
                b = srednia -1;
            }
            else
            {
                i = srednia +1;
            }
        }
        if (nums[i]>=target)
        {
            return i;
        }
        else 
        {
            return i+1;
        }

    }
};

File: MateuszSapala/35.Search_Insert_Position.kt
//https://leetcode.com/problems/search-insert-position/submissions/877566572/
class Solution {
    fun searchInsert(nums: IntArray, target: Int): Int {
        for (i in 0..nums.size - 1) {
            if (nums[i] >= target) {
                return i
            }
        }
        return nums.size
    }
}
